src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h

```diff
@@ -268,15 +347,18 @@ inline void saveRuptureFrontOutput(bool ruptureTimePending[misc::numPaddedPoints
  * param[in] slipRateMagnitude
  * param[in, out] peakSlipRate
  */
+template <RangeType Type = RangeType::CPU>
 inline void savePeakSlipRateOutput(real slipRateMagnitude[misc::numPaddedPoints],
-                                   real peakSlipRate[misc::numPaddedPoints]) {
+                                   real peakSlipRate[misc::numPaddedPoints],
+                                   unsigned startIndex = 0) {
+
+  using Range = typename NumPoints<Type>::Range;
 
-#ifdef ACL_DEVICE_OFFLOAD
-#pragma omp loop bind(parallel)
-#else
-#pragma omp simd
+#ifndef ACL_DEVICE
+  #pragma omp simd
 #endif
-  for (unsigned pointIndex = 0; pointIndex < misc::numPaddedPoints; pointIndex++) {
+  for (auto index = Range::start; index < Range::end; index += Range::step) {
+    auto pointIndex{startIndex + index};
```

Comment:
Also consider to put this into the loop header.

Reply:
discussed